controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/DeleteReaderGroupTask.java

```diff
@@ -0,0 +1,79 @@
+import io.pravega.controller.task.Stream.StreamMetadataTasks;
+import io.pravega.shared.NameUtils;
+import io.pravega.shared.controller.event.DeleteReaderGroupEvent;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Iterator;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ScheduledExecutorService;
+
+/**
+ * Request handler for executing a delete operation for a ReaderGroup.
+ */
+@Slf4j
+public class DeleteReaderGroupTask implements ReaderGroupTask<DeleteReaderGroupEvent> {
+
+    private final StreamMetadataStore streamMetadataStore;
+    private final StreamMetadataTasks streamMetadataTasks;
+    private final ScheduledExecutorService executor;
+
+    public DeleteReaderGroupTask(final StreamMetadataTasks streamMetaTasks,
+                                 final StreamMetadataStore streamMetaStore,
+                                 final ScheduledExecutorService executor) {
+        Preconditions.checkNotNull(streamMetaStore);
+        Preconditions.checkNotNull(streamMetaTasks);
+        Preconditions.checkNotNull(executor);
+        this.streamMetadataStore = streamMetaStore;
+        this.streamMetadataTasks = streamMetaTasks;
+        this.executor = executor;
+    }
+
+    @Override
+    public CompletableFuture<Void> execute(final DeleteReaderGroupEvent request) {
+      String scope = request.getScope();
+      String readerGroup = request.getRgName();
+      long requestId = request.getRequestId();
+      log.info("Inside Delete Event");
+      final RGOperationContext context = streamMetadataStore.createRGContext(scope, readerGroup);
+      return streamMetadataStore.getReaderGroupConfigRecord(scope, readerGroup, context, executor)
+               .thenCompose(configRecord -> {
+               if (!ReaderGroupConfig.StreamDataRetention.values()[configRecord.getObject().getRetentionTypeOrdinal()]
+                       .equals(ReaderGroupConfig.StreamDataRetention.NONE)) {
+               String scopedRGName = NameUtils.getScopedReaderGroupName(scope, readerGroup);
+               log.info("Inside Delete Event, found config");
+               // update Stream metadata tables, if RG is a Subscriber
+               Iterator<String> streamIter = configRecord.getObject().getStartingStreamCuts().keySet().iterator();
+               return Futures.loop(() -> streamIter.hasNext(), () -> {
+                       Stream stream = Stream.of(streamIter.next());
+                       return streamMetadataStore.deleteSubscriber(stream.getScope(),
+                               stream.getStreamName(), scopedRGName, null, executor);
+                       }, executor);
+               }
+               return CompletableFuture.completedFuture(null);
+               })
+               .thenCompose(v -> {
+                   log.info("Inside Delete Event, deleting metadata and stream");
+                   return streamMetadataStore.deleteStream(scope,
```

Comment:
again this is merely updating the metadata and not deleting the actual stream.. 
use streammetadatatasks.deletestream to actually delete the stream.
the caveat there is that this stream will be deleted asynchronously and this processing will wait on that to complete.. i dont particularly like that pattern. 

for that reason it may be preferrable to delete the RG stream outside the context of the event processing.. and then posting the event to perform unsubscriptions.

Reply:
Fixed to invoke streammetadatatasks.deleteStream()